src/nvim/ex_docmd.c

```diff
@@ -6948,10 +6948,39 @@ void free_cd_dir(void)
 
 #endif
 
+void apply_autocmd_dirchanged(char_u *new_dir, CdScope scope)
+{
+  dict_T *dict = get_vim_var_dict(VV_EVENT);
+  char buf[8];
+
+  switch (scope) {
+  case kCdScopeGlobal:
+    snprintf(buf, sizeof buf, "global");
+    break;
+  case kCdScopeTab:
+    snprintf(buf, sizeof buf, "tab");
+    break;
+  case kCdScopeWindow:
+    snprintf(buf, sizeof buf, "window");
+    break;
+  default:
```

Comment:
They do that so that if new values for the enum get added you'll see the warning about it being unhandled instead of it being swallowed by the `default` case.